docs/tutorials/data_provider.md

```diff
@@ -61,3 +61,100 @@ impl AdditiveIdentity {
     }
 }
 ```
+
+## Caching Data Provider
+
+It is not generally required to implement a cache, especially if using the BakedDataProvider. However, if you wish to implement a cache in your data pipeline, you can do so like this:
+
+```rust
+use icu_provider::hello_world::HelloWorldFormatter;
+use icu_provider::prelude::*;
+use icu::locid::locale;
+use lru::LruCache;
+use std::convert::TryInto;
+use std::sync::Mutex;
+use yoke::trait_hack::YokeTraitHack;
+use yoke::Yokeable;
+use zerofrom::ZeroFrom;
+
+pub struct MyLruDataCache<P> {
+    cache: Mutex<LruCache<(DataKey, DataLocale), AnyResponse>>,
+    provider: P,
+}
+
+impl<M, P> DataProvider<M> for MyLruDataCache<P>
+where
+    M: KeyedDataMarker + 'static,
+    M::Yokeable: ZeroFrom<'static, M::Yokeable>,
+    M::Yokeable: icu_provider::MaybeSendSync,
+    for<'a> YokeTraitHack<<M::Yokeable as Yokeable<'a>>::Output>: Clone,
+    P: DataProvider<M>,
+{
+    fn load(&self, req: DataRequest) -> Result<DataResponse<M>, DataError> {
+        let cache_key = (M::KEY, req.locale.clone());
+        {
+            let mut cache = self.cache.lock().unwrap();
+            if let Some(any_resp) = cache.get(&cache_key) {
```

Comment:
`any_response`

Reply:
`req` and `resp` are common variable names.

Reply:
The ones I usually see are `req` and `res`. But I don't think that logic extends to prefixing it. You're so close to writing the whole thing.

Reply:
The argument is named `req` so I'm being consistent with that. I changed `resp` to `res`. I disagree with the notion that the choice of prefix should affect whether or not the suffix is expanded.

Reply:
Below you have 
* `computed_resp`
* `computed_any_resp`

The former is 4 instead of 5 syllables, the latter 5 instead of 6. Why not `comp_any_resp` or `car`? Because that's confusing and we're not in C or Go. Same goes for the `resp` part imho.

Reply:
I'm fine with `comp_res` and `comp_any_res`. done.

Reply:
Not what I meant, I'm not fine with either of those and was using them as an example where you didn't feel the need to shorten the names. I think abbreviated variable names are one of the worst things one can do for readability (and that goes for `req` and `res` as well, although I can excuse the `req` parameter name as otherwise it gets repetitive with the type). `comp` for example can mean `compare`, `computed` (on that note, "computed" is not a very meaningful prefix), `competition`, `competitor`, ... . I don't want to read code to figure out what variable names mean, it should be the other way around, the variable name should tell me what the code means.

Reply:
I think I agree with you? I prefer `computed_res` because "comp" is very confusable, but "res" is not confusable, especially with the context of the variable named "req" and the type of the variable that says the fully spelled out "Response".

Reply:
`res` is as confusable as `comp`, and so is `resp`. `any_res` does not have an explicit type, and it's a variable, not a documented parameter like `req`. What do you gain by saving those 4 characters?

Also, why not just call `computed_res` `response`?

Reply:
`response` is not my first or second or third choice, but it's the best of the intersection of names we seem to think are acceptable. Done.

I also got rid of the `computed_any_res` variable.